### ISSUE TITLE
Typescript type checking

### ISSUE THREAD

**Issue by nchanged** [fuse-box/fuse-box#146]
Currently, FuseBox does not do type checking. For known reasons - Speed and caching involved. 
What could be done is a daemon, that runs in the background and reports errors.

What do you think?

**Comment by Grafikart** [comment]
I think this is a good idea. For instance I chose https://github.com/s-panferov/awesome-typescript-loader for webpack for this same reason. Typescript check is done on a separate process and won't block the build but will output an alert on the console (and display an overlay on the HTML)

![](https://pbs.twimg.com/media/CcfFabVWIAEzUKb.png)

**Comment by etienne-dldc** [comment]
Good idea 👍  But what do you mean exactly by "daemon" ?

**Comment by donskifarrell** [comment]
We also use TSLint. It would be great to have it report back in a similar fashion to type checking (in the terminal at least)

**Comment by devmondo** [comment]
@donskifarrell some generous member on gitter is working on TS Lint plugin, and it is almost done  :)

**Comment by vegarringdal** [comment]
This was just what I was looking for, how is the TS Lint plugin going ?

**Comment by RPDeshaies** [comment]
I also, would be interested by a TSLint plugin :)

**Comment by vegarringdal** [comment]
> some generous member on gitter is working on TS Lint plugin, and it is almost done :)

@devmondo, do you ahve any status on this one, would have been great to get errors up when its compiling..

**Comment by devmondo** [comment]
@vegarringdal the sad part is that i don't remember who is the member working on this, so please if you see this anonymous person give us some feedback :)

but we think we should enable some kind of demon as suggested above, same way IDE's and Editors do it like webstorm and VSCode, what do you guys think ?

**Comment by nchanged** [comment]
https://github.com/fuse-box/fuse-box/issues/330

**Comment by ma2ciek** [comment]
For me it's a must-have option for TS compiler. In a bigger than 20-file size project you won't be able to track every file simultaneously to find IDE errors and you'll loose the biggest TS advantage. +100 for this issue.

**Comment by unlight** [comment]
Not an issue for me, I do typechecking in separate task with lint.

**Comment by ma2ciek** [comment]
@unlight Are you using just tsc for this purpose? Maybe I exaggerated this problem.

**Comment by unlight** [comment]
Not only tsc.
I just mean that typechecking task is runnning in parallel dev server and other stuff.
See example [here](https://github.com/unlight/angular-fusebox-seed/blob/f1583512404e5302dbebbc917c6ade19add0f987/package.json#L23-L25)

**Comment by vegarringdal** [comment]
I guess we could make a simple plugin for this. (typechecking)
Would we like it to post all errors when 1 file is changed or just the errors in file changed?

**Comment by ma2ciek** [comment]
Thanks @unlight for the clarification. I didn't know these packages.

**Comment by nchanged** [comment]
@vegarringdal it has to compile the entire project ... . To ensure that everything is intact.
It's like
```
tsc --watch
```

**Comment by vegarringdal** [comment]
@nchanged 

K, Im still gonna try and make a plugin using the typescript api, if I fail, then its not like we lost something 😂

**Comment by vegarringdal** [comment]
Silly output, but something...
![image](https://cloud.githubusercontent.com/assets/2901416/23554807/dc7eff4a-0026-11e7-92cd-330e698ef0fa.png)

**Comment by vegarringdal** [comment]
Its all about being a great artist (copy others code) and failing a lot 😂

**Comment by vegarringdal** [comment]
Latest progress on this, made it into a plugin


![image](https://cloud.githubusercontent.com/assets/2901416/23560144/b8515402-0039-11e7-87c4-c2d9cd0b7fd2.png)

**Comment by vegarringdal** [comment]
Hi all... made this one :-)

https://github.com/fuse-box/fuse-box-typechecker

**Comment by vegarringdal** [comment]
Updated https://github.com/fuse-box/fuse-box-typechecker to run the type checking in own thread so it does not block bundle/devserver

Anyone want to give it a try  and give some feedback ?

**Comment by RPDeshaies** [comment]
Will install on monday for sure.

**Comment by vegarringdal** [comment]
@RPDeshaies Great, its not perfect, but it helps

**Comment by RPDeshaies** [comment]
It will probably be better then to rely only on VSCode and precommit rules... :) ! Thanks for all your effort !

**Comment by RPDeshaies** [comment]
I'm having some problems with it. I'm going to create issues to not hijack the thread

**Comment by nchanged** [comment]
all here
https://github.com/fuse-box/fuse-box-typechecker

**Comment by 0xorial** [comment]
This should be mentioned in the docs.
